### ISSUE TITLE
Fix Less mixins with !important and shared-line comment autofix

### ISSUE THREAD

**Issue by mahboob-awan** [stylelint/stylelint#2826]
> Bug

stylelint's --fix mess with LESS mixins's comment & !important

> Which rule, if any, is this issue related to?

No idea but none of the _(Autofixable)_ is part of stylelint rules

> What LESS is needed to reproduce this issue?

Following change will explain before/after state:

```less
//Example#1 - change position of !important
 .mq-hide-small {
 	.mq-small( {
-		display: none !important;
-	} );
+		display: none;
+	} ) !important;
 }

//Example#2 - remove comments from within LESS mixin
 	.mq-large( {
-		z-index: 1; // comments
+		z-index: 1; 
 		padding: 0;
 	} );

//WHERE mq-large is:
.mq-large(@rules) {
	@media only screen and (min-width: (@media-medium + @media-padding)) {
		@rules();
	}
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
				"rules": {
					//================================= Possible Errors
					"at-rule-no-unknown": [true, {ignoreAtRules:["rules"]}],
					"block-no-empty": true,
					"color-no-invalid-hex": true,
					"comment-no-empty": true,
					"declaration-block-no-duplicate-properties": [ true, {
						ignore: ["consecutive-duplicates-with-different-values"],
					} ],
					"declaration-block-no-redundant-longhand-properties": true,
					"declaration-block-no-shorthand-property-overrides": true,
					"font-family-no-duplicate-names": true,
					"function-calc-no-unspaced-operator": true,
					"function-linear-gradient-no-nonstandard-direction": true,
					"keyframe-declaration-no-important": true,
					"media-feature-name-no-unknown": true,
					"no-empty-source": true,
					"no-extra-semicolons": true,
					"no-invalid-double-slash-comments": true,
					"property-no-unknown": true,
					"selector-pseudo-class-no-unknown": true,
					"selector-pseudo-element-no-unknown": true,
					"selector-type-no-unknown": [true, {ignoreTypes: ["/^ns/"]}],
					"shorthand-property-no-redundant-values": true,
					"string-no-newline": true,
					"unit-no-unknown": true,

					//================================= Limit language features
					// Color
					"color-named": "never",

					//Values
					"value-list-max-empty-lines": 0,

					// Selectors
					"selector-max-empty-lines": 1,

					//NUMBER
					"number-max-precision": 3,

					//UNIT
					"unit-blacklist": [
						"rem",
						"vmin",
						"vmax",
						"ex",
						"ch" // [1]
					],

					//PROPERTY
					"property-blacklist" : ["font-weight"],

					//=================================Stylistic issues
					//Font-family
					"font-family-name-quotes": "always-where-recommended",

					//Functions
					"function-name-case": "lower",
					"function-url-quotes": "always",

					// Blocks
					"block-opening-brace-newline-after": "always",

					//LENGTH
					"length-zero-no-unit": true,

					//NUMBER
					"number-no-trailing-zeros": true,
					"number-leading-zero": "never",

					//PROPERTY
					"property-case": "lower"
				}
```

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Node API

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, It's related LESS mixin.

> What did you expect to happen?

LESS should not mess-up or at least avoid from transforming into invalid LESS code.

> What actually happened (e.g. what warnings or errors you are getting)?

No warning

Please ask, if more detail required. Thanks

**Comment by jeddy3** [comment]
@mahboob-awan Thanks for the report and for using the template. This _could_ be an issue with either the parser or stringifier of [`postcss-less`](https://github.com/shellscape/postcss-less). Please consider digging deeper and contributing a fix if you'd like to see this issue closed, as the stylelint team is currently focused on standard CSS syntax.

**Comment by federinik** [comment]
+1

**Comment by jwilsson** [comment]
This one seems to be fixed with `stylelint@9.7.0` and `postcss-less@3.0.0`.
